nengo_dl/tests/test_simulator.py

```diff
@@ -835,7 +838,9 @@ def test_profile(Simulator, mode, tmp_path):
     with Simulator(net) as sim:
         # note: TensorFlow bug if using profile_batch=1, see
         # https://github.com/tensorflow/tensorflow/issues/37543
-        callback = callbacks.TensorBoard(log_dir=tmp_path / "profile", profile_batch=2)
+        callback = callbacks.TensorBoard(
+            log_dir=str(tmp_path / "profile"), profile_batch=2
```

Comment:
Why do we need an explicit `str` call here, but not earlier in this file when we create a `tf.keras.callbacks.TensorBoard`? (`callbacks.TensorBoard` uses the same constructor, so I'd think they'd be the same.)

Reply:
Hmm I'm not sure. Generally I started by just passing in a Path wherever I could, and only converted it to a string if that failed. But I don't know why this one would have failed and the other didn't, maybe I just missed this one.

Reply:
Tried removing the `str` and it fails. No idea why it fails here and not in the other place, but either way it's all on the TensorFlow side of things so all we can do is cast the paths to strings where required.